src/quicklist.c

```diff
@@ -1244,17 +1244,17 @@ int quicklistIndex(const quicklist *quicklist, const long long idx,
     initEntry(entry);
     entry->quicklist = quicklist;
 
-    if (!forward) {
-        index = (-idx) - 1;
-        n = quicklist->tail;
-    } else {
-        index = idx;
-        n = quicklist->head;
-    }
-
+    index = forward ? idx : (-idx) - 1;
     if (index >= quicklist->count)
         return 0;
 
+    /* Change direction if the other way is shorter. */
+    if (index > (quicklist->count - 1) / 2) {
+        forward = !forward;
```

Comment:
@zuiderkwast I got an error when running REDIS_TEST (`itrprintr` and `itrprintr_rev`), because you changed the direction of the traversal and ended up with the wrong number of times for quicklsitNext.
Now the code doesn't use the entry fetched by quicklistIndex for traversal, but I think we still need to support external control of the traversal direction.

Reply:
Thanks for testing with REDIS_TEST! I forgot to do it. :facepalm: This needs to be fixed before merge.

Reply:
Solved. The problem was that `entry->offset` was not set correctly in `quicklistIndex()` according to the requested direction.